### PR TITLE
[docs] Add doc for recording lifted details

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
Signed-off-by: changzhen <changzhen5@huawei.com>

**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

We record a list of all files lifted from other projects for unified management.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
/hold
wait for #1460

**Comment by XiShanYongYe-Chang** [comment]
/hold
wait for #1460

File: pkg/util/lifted/lifted_record_list.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package lifted
```
Comment (RainbowMango) [inline pkg/util/lifted/lifted_record_list.go:1]:
```suggestion
// Package lifted contains the files lifted from other projects.
package lifted
```

And rename the file name to `docs.go`

File: pkg/util/lifted/lifted_record_list.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+// This file records all files listed from other projects.
```
Comment (RainbowMango) [inline pkg/util/lifted/lifted_record_list.go:3]:
```suggestion
```

File: pkg/util/lifted/lifted_record_list.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+/*
+| karmada file path        | other project file path                                                                                                  | const/var/type/func                     | changed |
```
Comment (RainbowMango) [inline pkg/util/lifted/lifted_record_list.go:6]:
```suggestion
| lifted file        | source file                                                                                                  | const/var/type/func                     | changed |
```

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**